src/views/recordTable.component.tsx

```diff
@@ -103,100 +88,63 @@ const RecordTable = React.memo(
-        keys.forEach((key: string) => {
-          const channel: Channel = record.channels[key];
-          const channelData = channel.data;
-          recordRow[key] = channelData;
-        });
-
-        newData.push(recordRow);
-      });
-
-      setParsedData(newData);
-      return newData;
-    };
-
-    React.useEffect(() => {
-      if (data && !dataLoading && !channelsLoading) {
-        const parsedData: RecordRow[] = parseData(data);
-        const newAvailableColumns = constructColumns(parsedData);
-        setAvailableColumns(newAvailableColumns);
-        setColumnsLoaded(true);
-      } else {
-        setColumnsLoaded(false);
-      }
-      // eslint-disable-next-line react-hooks/exhaustive-deps
-    }, [data, dataLoading, channelsLoading]);
-
-    React.useEffect(() => {
-      setQueryParams({
-        page: page,
-        sort: sort,
-        dateRange: dateRange,
-      });
-    }, [page, sort, dateRange]);
-
-    const onPageChange = (page: number) => {
-      setPage(page);
-    };
-
-    const handleSort = (column: string, order: Order | null) => {
-      let newSort = sort;
-      if (order !== null) {
-        newSort = {
-          ...newSort,
-          [column]: order,
-        };
-      } else {
-        const { [column]: order, ...rest } = newSort;
-        newSort = {
-          ...rest,
-        };
-      }
-      setSort(newSort);
-    };
-
-    const handleDateTimeChange = (
-      range: 'fromDate' | 'toDate',
-      date?: string
-    ) => {
-      setDateRange({ ...dateRange, [range]: date });
-    };
-
-    return (
-      <div>
-        <DateTimeInputBox onChange={handleDateTimeChange} />
-        <br />
-        <Table
-          data={parsedData}
-          availableColumns={availableColumns}
-          totalDataCount={count ?? 0}
-          page={page}
-          loadedData={columnsLoaded}
-          loadedCount={!countLoading}
-          resultsPerPage={resultsPerPage}
-          onPageChange={onPageChange}
-          sort={sort}
-          onSort={handleSort}
-        />
-      </div>
-    );
-  }
-);
+    },
+    [channels]
+  );
+
+  React.useEffect(() => {
+    if (data && !channelsLoading && !columnsLoaded) {
+      // columns normally don't reload each time the data changes
+      const newAvailableColumns = constructColumns(data);
+      setAvailableColumns(newAvailableColumns);
+      setColumnsLoaded(true);
+    }
+  }, [data, channelsLoading, columnsLoaded, dispatch, constructColumns]);
+
+  const onPageChange = React.useCallback(
+    (page: number) => {
+      dispatch(changePage(page));
+    },
+    [dispatch]
+  );
+
+  const onResultsPerPageChange = React.useCallback(
+    (resultsPerPage: number) => {
+      dispatch(changeResultsPerPage(resultsPerPage));
+    },
+    [dispatch]
+  );
+
+  const handleSort = React.useCallback(
+    (column: string, order: Order | null) => {
+      dispatch(changeSort({ column, order }));
+    },
+    [dispatch]
+  );
```

Comment:
I personally think these could be moved to table? I don't wanna bloat the file too much but they already have some other event handlers like `handleColumnClose`, `handleColumnWordWrapToggle`, etc.

Reply:
yeah, I wanted some opinions on whether to have all the redux stuff in `Table` and have `RecordTable` just handle the data querying perhaps or to have `Table` be unconnected and put the redux stuff in `RecordTable` and pass it down as props to `Table`. As you can see it created some weirdness in the testing as some stuff was in one place and some stuff is in another...

Reply:
Yeah it's all about figuring out what makes sense to put where, like when I originally had the checkboxes in with the table component because I thought that made sense at the time but now they've been moved back to recordTable, I don't know what I was thinking! It's much better there. So yeah figuring out where things belong can be a tough job

Reply:
I've gone for the other way round - I'm extracting all the redux & react-query stuff to `RecordTable`, so `Table` is "dumb" and just deals with `react-table` and styling. I think this separates things logically